src/memtable.rs

```diff
@@ -444,7 +441,7 @@ impl<E: Message + Clone, W: EntryExt<E>> MemTable<E, W> {
             .rev()
             .find(|e| e.file_num <= latest_rewrite);
         if let (Some(begin), Some(end)) = (begin, end) {
-            if begin.index < end.index {
+            if begin.index <= end.index {
```

Comment:
What's the corner case of using `<=` instead of `<`

Reply:
If `begin.index == end.index`, there is one and only one log entry in the append queue. If we don't rewrite it, the append queue can't be purged on time. I meet the bug yesterday.